src/server/scripts/Custom/SkipDK.cpp

```diff
@@ -340,6 +155,43 @@ class Trinitycore_optional_deathknight_skip : public CreatureScript
     }
 };
 
+void Trinitycore_skip_deathknight_HandleSkip(Player* player)
+{
+    //Not sure where DKs were supposed to pick this up from, leaving as the one manual add
+    player->AddItem(6948, true); //Hearthstone
+
+    // these are all the starter quests that award talent points, quest items, or spells
+    int STARTER_QUESTS[39] = { 12593, 12619, 12842, 12848, 12636, 12641, 12657, 12678, 12679, 12680, 12687, 12698, 12701, 12706, 12716, 12719, 12720, 12722, 12724, 12725, 12727, 12733, 12739, 12742, 12743, 12744, 12745, 12746, 12747, 12748, 12749, 12750, 12751, 12754, 12755, 12756, 12757, 12779, 12801 };
+
+    for (auto questId : STARTER_QUESTS)
+    {
+        if (player->GetQuestStatus(questId) == QUEST_STATUS_NONE)
+        {
+            player->AddQuest(sObjectMgr->GetQuestTemplate(questId), nullptr);
+            player->RewardQuest(sObjectMgr->GetQuestTemplate(questId), false, player);
+        }
+    }
+
+    int DKL = sConfigMgr->GetFloatDefault("Skip.Deathknight.Start.Level", 58);
+    if (player->GetLevel() <= DKL)
+    {
+        //GiveLevel updates character properties more thoroughly than SetLevel
+        player->GiveLevel(DKL);
+    }
+
+    //Don't need to save all players, just current
+    player->SaveToDB();
```

Comment:
this part can work, the all save is being debated in being removed as there are some reports of deadlocks happening with high population servers of 500 or more players.